dxlepoclient/client.py

```diff
@@ -245,14 +245,14 @@ def _sync_request(dxl_client, request, response_timeout, payload_dict):
         res = dxl_client.sync_request(request, timeout=response_timeout)
 
         # Return a dictionary corresponding to the response payload
-        if res.message_type != Message.MESSAGE_TYPE_ERROR:
-            ret_val = MessageUtils.decode_payload(res)
-            # Display the response
-            logger.debug("Response:\n%s", ret_val)
-            return ret_val
-        else:
-            raise Exception("Error: " + res.error_message + " (" + str(
-                res.error_code) + ")")
+        if res.message_type == Message.MESSAGE_TYPE_ERROR:
+            Exception("Error: " + res.error_message + " (" + str(res.error_code) + ")")
```

Comment:
I think this line should have a `raise` keyword in front of it.

Also, it may now make more sense to move the `# Return a dictionary...` line down past the `if` block since the logic to do a return is only followed when the `if` block is not entered.